src/main/java/org/spongepowered/api/util/command/CommandResult.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util.command;
+
+import com.google.common.base.Optional;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents the result of a command in Sponge.
+ */
+public class CommandResult {
+
+    private final boolean processed;
+    private final Optional<Integer> successCount;
+    private final Optional<Integer> affectedBlocks;
+    private final Optional<Integer> affectedEntities;
+    private final Optional<Integer> affectedItems;
+    private final Optional<Integer> queryResult;
+    private final Map<String, Object> resultInfo;
+
+    /**
+     * Constructs a new command result.
+     * 
+     * @param processed True if the command was processed
+     * @param successCount The success count
+     * @param affectedBlocks The number of affected blocks
+     * @param affectedEntities The number of affected entities
+     * @param affectedItems The number of affected items
+     * @param queryResult The query result
+     * @param resultInfo A map storing additional information
+     */
+    protected CommandResult(boolean processed, @Nullable Integer successCount, @Nullable Integer affectedBlocks, @Nullable Integer affectedEntities,
```

Comment:
This should be private or package-local (no spec) or this class should be marked final. We only want people instantiating it through the builder